DigitalLearningSolutions.Web/Controllers/Register/RegisterDelegateByCentreController.cs

```diff
@@ -203,7 +208,41 @@ public IActionResult Summary()
         [HttpPost]
         public IActionResult Summary(SummaryViewModel model)
         {
-            return new OkResult();
+            var data = TempData.Peek<DelegateRegistrationByCentreData>()!;
+
+            var candidateNumber =
+                registrationService.RegisterDelegateByCentre(
+                    RegistrationMappingHelper.MapToDelegateRegistrationModel(data)
+                );
+
+            switch (candidateNumber)
+            {
+                case "-1":
+                    return RedirectToAction("Error", "LearningSolutions");
+                case "-4":
+                    return RedirectToAction("Index");
+            }
+
+            TempData.Clear();
+            TempData.Add("delegateNumber", candidateNumber);
+            TempData.Add("emailSent", data.ShouldSendEmail);
+            TempData.Add("passwordSet", data.IsPasswordSet);
+            return RedirectToAction("Confirmation");
+        }
+
+        [HttpGet]
+        public IActionResult Confirmation()
+        {
+            var delegateNumber = (string?)TempData.Peek("delegateNumber");
+            var emailSent = (bool)TempData.Peek("emailSent");
+            var passwordSet = (bool)TempData.Peek("passwordSet");

```

Comment:
Should we clear temp data on this final page (or just Get instead of Peek which will be equivalent)?